### ISSUE TITLE
syscall: Segfault on Select on ARM64 and MIPS64

### ISSUE THREAD

**Issue by zx2c4** [golang/go#24950]
The `select` syscall takes a timespec param. If it contains a time of zero, select returns immediately. If it contains a positive amount of time, select times out after that amount of time. If it is null, it never times out. Sometimes this latter behavior is desirable. It's well documented, mentioned in the man page, and used widely. However, https://go-review.googlesource.com/c/sys/+/70610 introduced a regression whereby Go will segfault if the timespec is nil, on ARM64 and MIPS64.

The current code for those platforms is:

```go
func Select(nfd int, r *FdSet, w *FdSet, e *FdSet, timeout *Timeval) (n int, err error) {
        ts := Timespec{Sec: timeout.Sec, Nsec: timeout.Usec * 1000}
        return pselect(nfd, r, w, e, &ts, nil)
}
```

The correct code should be:

```go
func Select(nfd int, r *FdSet, w *FdSet, e *FdSet, timeout *Timeval) (n int, err error) {
        if timeout != nil {
                ts := Timespec{Sec: timeout.Sec, Nsec: timeout.Usec * 1000}
                return pselect(nfd, r, w, e, &ts, nil)
        } else {
                return pselect(nfd, r, w, e, nil, nil)
        }
}
```

Currently I'm having to patch my local installation of Go to produce working executables for Android devices.

CC: @ianlancetaylor @tklauser

**Comment by ianlancetaylor** [comment]
As I replied earlier on this CL, I believe this was fixed by https://golang.org/cl/97820.

**Comment by ianlancetaylor** [comment]
Oh, sorry, you are talking about syscall, not x/sys/unix.  First: use x/sys/unix.  Second: in syscall this was fixed by https://golang.org/cl/97819, so the fix will be in 1.11.  I don't think we need to backport this to 1.10 because: use x/sys/unix.

**Comment by zx2c4** [comment]
Oh, great. Plans to backport this to 1.10? Or will I need to wait until 1.11?
